src/JsonSchema/TypeFactory.php

```diff
@@ -29,6 +30,19 @@ final class TypeFactory implements TypeFactoryInterface
      */
     private $schemaFactory;
 
+    /**
+     * @var array
+     */
+    private $resourceClasses;
+
+    /**
+     * TypeFactory constructor.
+     */
+    public function __construct(ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory = null)
+    {
+        $this->resourceClasses = $resourceNameCollectionFactory ? iterator_to_array($resourceNameCollectionFactory->create()) : [];
```

Comment:
Can you delay the call to `iterator_to_array` by moving it outside of the constructor, in case the service is constructed but not used, for performance reasons.

Reply:
alright :-)